### PR TITLE
Clarify usage for the git backend's :host and :repo keywords

### PR DESCRIPTION
**Pull request by hristoast**
As we discussed in #505 I've added a bit more clarification about these keywords. I wasn't sure if the code snippets fit but I decided to just go with them anyways. Let me know if you'd prefer this PR not have them!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2141,8 +2141,19 @@ These are the keywords meaningful for the `git` backend:
...
 * `:repo`: the clone URL for the Git repository.
 * `:host`: either nil or one of the symbols `github`, `gitlab`,
-  `bitbucket`. If non-nil, then `:repo` should just be a string
-  "username/repo", and the URL is constructed automatically.
+  `bitbucket`.
+    * If nil, then `:repo` should be a string which is the full URL of
+    the target repository. For example:
+
+            ( :package "package" :host nil :type git
+              :repo "http://myhost.tld/repo")
```
Comment (progfolio) [inline README.md:2149]: Could you please change these examples to use the fenced codeblock syntax?
e.g.
     ```emacs-lisp
    (example... ;etc
    )
    ```

**Comment by progfolio** [comment]
Don't worry about those two CI failures. They're due to the concurrency limit with our CI service.

**Comment by progfolio** [comment]
Thanks again.

**Comment by raxod502** [comment]
Huh... is CircleCI reporting build failures just because of the concurrency limit? If so, that seems pretty bad: I would expect it to simply take longer to report a result.

If this behavior happens on a regular basis, we might want to try out GitHub Actions, which has worked well for me in the past and has been simple to set up.

**Comment by progfolio** [comment]
I believe what happens is the job is started concurrently, then when the limit is hit the job fails due to a network timeout e.g.:

```
error pulling image configuration: Get https://production.cloudflare.docker.com/registry-v2/docker/registry/v2/blobs/sha256/fa/fa24704bae4cb348ddffd7af0137f9e3755504f2ad4b0965f88a69b04d23f22e/data?verify=1616250992-AkGSZTb1X2qfDTmOeeiFr2YvtI0%3D: net/http: TLS handshake timeout
```

**Comment by raxod502** [comment]
I am pretty sure that is not an error on the side of CircleCI. I have seen the concurrency limit warning you mentioned, and I am pretty sure it just means that some of the jobs were queued before being executed. That error message seems to indicate a configuration error on Docker Hub's side. Even if we were hitting the Docker Hub free tier rate limit which was instituted recently, we should be getting an API error back, not a TLS handshake timeout. If it happens on a regular basis, and this error hasn't been well-discussed online, I'd reach out to Docker Hub, to be honest.

**Pull request merged**